kernel/src/process.rs

```diff
@@ -121,30 +230,19 @@ pub enum IPCType {
 }
 
 #[derive(Copy, Clone)]
-crate enum Task {
+pub enum Task {
     FunctionCall(FunctionCall),
     IPC((AppId, IPCType)),
 }
 
+/// Struct that defines a callback that can be passed to a process.
 #[derive(Copy, Clone, Debug)]
-crate struct FunctionCall {
-    crate r0: usize,
-    crate r1: usize,
-    crate r2: usize,
-    crate r3: usize,
-    crate pc: usize,
-}
-
-#[derive(Default)]
-struct StoredRegs {
-    r4: usize,
-    r5: usize,
-    r6: usize,
-    r7: usize,
-    r8: usize,
-    r9: usize,
-    r10: usize,
-    r11: usize,
+pub struct FunctionCall {
+    pub r0: usize,
+    pub r1: usize,
+    pub r2: usize,
+    pub r3: usize,
+    pub pc: usize,
```

Comment:
The concept of a function call accepting arguments as register0, ... is probably not universal. Maybe rename these to `argument0`, `argument1`, ... which is their purpose anyways. And `pc` could be called `functionAddress` or similar.

Reply:
While I agree that `arugmentN` is a clearer name, I actually looked into this the first time this came around -- more on the bend of understanding if 4 arguments was a reasonable expectation as we look at other architectures. Turns out that syscall argument passing via registers is fairly universal, and ARM is actually the odd fellow out with 4 as a lowball number of supported arguments.

Reply:
So, leave it?

Reply:
I think `argumentN` is a better name, my comment was intended to address Branden's "... is probably not universal"

Reply:
yeah, `argumentN` is better. a comment could say something like ("These will typically be passed in the first few registers, but it's architecture specific").